pkg/services/store/service.go

```diff
@@ -117,48 +120,43 @@ func (s *standardStorageService) Read(ctx context.Context, user *models.SignedIn
 	return s.tree.GetFile(ctx, getOrgId(user), path)
 }
 
-func isFileTypeValid(filetype string) bool {
-	if (filetype == "image/jpeg") || (filetype == "image/jpg") || (filetype == "image/gif") || (filetype == "image/png") || (filetype == "image/webp") {
-		return true
-	}
-	return false
-}
-
 type UploadRequest struct {
 	Contents           []byte
-	MimeType           string
+	MimeType           string // TODO: remove MimeType from the struct once we can infer it from file contents
```

Comment:
?? didn't we set the value from:
```
mimeType := http.DetectContentType(data)
```

Reply:
There are four ways we can infer the data type

1. the file extension
2. the mime type header
3. the content, via `http.DetectContentType(data)`
4. the content, via something else

1 & 2 can be easily spoofed. 3 doesn't always work (e.g. for SVGs) and its based on one spec so that won't change - https://mimesniff.spec.whatwg.org/. 

I don't think we want to change client's file extension, so my current idea is to 
1. infer the mime type from contents  based on something that always works, and then
2.  fail validation if inferred mime type doesn't match with the extension. I

I would completely ignore the client-passed mime-type header.